app/models/volunteer.rb

```diff
@@ -70,31 +70,43 @@ class Volunteer < ApplicationRecord
     if: :external?,
     unless: proc { |user| user.deleted? }
 
-  default_scope { order(created_at: :desc) }
+  scope :default_order, (-> { order(created_at: :desc) })
 
-  scope :created_between, ->(start_date, end_date) { where(created_at: start_date..end_date) }
+  scope :created_between, lambda { |start_date, end_date|
+    created_after(start_date).created_before(end_date)
+  }
   scope :created_before, ->(max_time) { where('volunteers.created_at < ?', max_time) }
   scope :created_after, ->(min_time) { where('volunteers.created_at > ?', min_time) }
 
   scope :with_hours, (-> { joins(:hours) })
   scope :with_assignments, (-> { joins(:assignments) })
+  scope :with_group_assignments, (-> { joins(:group_assignments) })
   scope :without_assignment, (-> { left_outer_joins(:assignments).where(assignments: { id: nil }) })
+  scope :without_group_assignments, lambda {
+    left_outer_joins(:group_assignments).where(group_assignments: { id: nil })
+  }
 
-  scope :with_active_assignments, (-> { joins(:assignments).merge(Assignment.active) })
   scope :with_active_assignments_between, lambda { |start_date, end_date|
     joins(:assignments).merge(Assignment.active_between(start_date, end_date))
   }
+  scope :with_active_group_assignments_between, lambda { |start_date, end_date|
+    joins(:group_assignments).merge(GroupAssignment.active_between(start_date, end_date))
+  }
+
+  scope :with_active_assignments, (-> { joins(:assignments).merge(Assignment.active) })
+  scope :with_active_group_offers, (-> { joins(:group_offers).merge(GroupOffer.active) })
```

Comment:
It is possible though that a volunteer is removed from any group offer they were in and the group offer(s) remain(s) active.